engine/src/main/java/com/redhat/cloud/notifications/transformers/BaseTransformer.java

```diff
@@ -40,17 +45,61 @@ public JsonObject toJsonObject(final Action action) {
             EVENTS,
             new JsonArray(
                 action.getEvents().stream().map(
-                    event -> Map.of(
-                        METADATA, JsonObject.mapFrom(event.getMetadata()),
-                        PAYLOAD, JsonObject.mapFrom(event.getPayload())
+                    ev -> Map.of(
+                        METADATA, JsonObject.mapFrom(ev.getMetadata()),
+                        PAYLOAD, JsonObject.mapFrom(ev.getPayload())
                     )
                 ).collect(Collectors.toList())
             )
         );
         message.put(ORG_ID, action.getOrgId());
         message.put(TIMESTAMP, action.getTimestamp().toString());
 
+        // Since "insights-notifications-schemas-java:0.20".
+        this.appendSourceToMessage(message, event);
+
         return message;
     }
 
+    /**
+     * Appends the display names of the "application", "bundle" and "event
+     * type" in the "source" key of the payload. If the three of them are not
+     * present or, they are blank, then the key is not included in the payload.
+     * @param message the JSON object to append the "source" key to.
+     * @param event the event to get the display names from.
+     */
+    private void appendSourceToMessage(final JsonObject message, final Event event) {
+        final JsonObject source = new JsonObject();
+
+        JsonObject sourceAppDisplayName = null;
+        final String appDisplayName = event.getApplicationDisplayName();
+        if (appDisplayName != null && !appDisplayName.isBlank()) {
```

Comment:
IIRC all our event-types/apps/bundles should have a display name all the times- Is there a specific case where it does not?